src/kv/idbstore.rs

```diff
@@ -94,18 +94,10 @@ impl IdbStore {
         };
         let request = factory.open(name)?;
         let (callback, receiver) = IdbStore::oneshot_callback(logger.clone());
-        let request_copy = request.clone();
         let closure_logger = logger.clone();
-        let onupgradeneeded = Closure::once(move |_event: web_sys::IdbVersionChangeEvent| {
-            let result = match request_copy.result() {
-                Ok(r) => r,
-                Err(e) => {
-                    closure_logger.error(format!("Error before ugradeneeded: {:?}", e));
-                    return;
-                }
-            };
-            let db = web_sys::IdbDatabase::unchecked_from_js(result);
-
+        let onupgradeneeded = Closure::once(move |event: web_sys::IdbVersionChangeEvent| {
+            let target = event.target().unwrap();
```

Comment:
Don't panic. But here in handling some random event, we can't return an error code. So `error!` maybe?

Reply:
I should not have changed this function. I'm reverting that